### PR TITLE
[quest] fixes for 12979

### LINKED ISSUE #4325: Missing objective data for quest 12979 and objective Dark Armor Sample
Missing objective data for quest 12979 and objective Dark Armor Sample
Quest started by https://www.wowhead.com/wotlk/item=42203/dark-armor-plate
Item dropped by https://www.wowhead.com/wotlk/npc=29380/stormforged-war-golem

Questie version 7.3.0 - no updates pending
Wow version: wotlk classic - 3.4.0.45854

### PR DESCRIPTION
**Pull request by dscreet**
## Issue references

Fixes #4325

## Proposed changes

- Added objective data for quest

### CODE REVIEW

File: Database/Corrections/Wotlk/wotlkQuestFixes.lua [hunk 1]
```diff
@@ -1036,6 +1036,9 @@ function QuestieWotlkQuestFixes:Load()
             [questKeys.name] = "Hodir's Call",
             [questKeys.objectives] = {{{29974,"Niffelem Forefather freed"},},nil,nil,nil,{{{30144,30135},30144,},},},
         },
+        [12979] = {
+			[questKeys.objectives] = {nil,nil,{{42204},},nil,nil,},
```
Comment (Yttrium-tYcLief) [inline Database/Corrections/Wotlk/wotlkQuestFixes.lua:1040]: This is fine, but just for future reference you don't need to `nil` the rest of the table *after* your data. The preceding `nil`s are to ensure your data goes into the 3rd parameter (itemObjective), anything after that is needless (but legal) unless you need to add data to a later parameter.

The very act of re-defining the table will wipe out any existing data in `questKeys.objectives`.

File: Database/Corrections/Wotlk/wotlkItemFixes.lua [hunk 2]
```diff
@@ -206,6 +206,10 @@ function QuestieWotlkItemFixes:Load()
         [41399] = {
             [itemKeys.npcDrops] = {},
         },
+        [42203] = {
+            [itemKeys.npcDrops] = {29380},
+            [itemKeys.startQuest] = 12979,
```
Comment (Yttrium-tYcLief) [inline Database/Corrections/Wotlk/wotlkItemFixes.lua:211]: Does defining `itemKeys.startQuest` fix the chat prompt ("Picked up X item which starts Y quest!")? I've noticed that none of the WotLK quest-start items print that text, and as far as I can see they're missing this `itemKey`.

**Comment by Yttrium-tYcLief** [comment]
Added a few comments but overall looks good to me!

File: Database/Corrections/Wotlk/wotlkItemFixes.lua (hunk 2 above)
Comment (BreakBB) [reply Database/Corrections/Wotlk/wotlkItemFixes.lua:211]: Yes, that is absolute true. Good catch @dscreet 💪🏻 I'll see if we can script these missing fields into the other quest starting items.

**Review by BreakBB — APPROVED** [review]

**Comment by BreakBB** [comment]
Hey @dscreet this looks correct to me 👍🏻 Thank you very much for taking a look into it and providing this PR 🥳

File: Database/Corrections/Wotlk/wotlkQuestFixes.lua (hunk 1 above)
Comment (dscreet) [reply Database/Corrections/Wotlk/wotlkQuestFixes.lua:1040]: Gotcha, thanks.

File: Database/Corrections/Wotlk/wotlkItemFixes.lua (hunk 2 above)
Comment (dscreet) [reply Database/Corrections/Wotlk/wotlkItemFixes.lua:211]: Didn't know that was the case haha. I just double-checked all fields and saw this was missing. Gd to know.

**Pull request merged**